CDCgov/prime-reportstream
# 9411: Performance improvements for the STLT dashboard apis

This PR provides some performance updates to the new STLT dashboard apis

Test Steps:
1. Make sure that the the APIs for deliveries and submitters still load correctly

```sh
curl --location 'http://localhost:7071/api/v1/receivers/ignore.HL7/deliveries/submitters/search' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer dummy' \
--data '{
    "sort": {
        "direction": "ASC",
        "property": "sort_id"
    },
    "pagination": {
        "page": 1,
        "limit": 10
    },
    "filters": [
        {
            "value": "2023-05-21T00:00:00+00:00",
            "filterName": "UNTIL"
        }
    ]
}'
```

```sh
curl --location 'http://localhost:7071/api/v1/receivers/ignore.HL7/deliveries' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer dummy' \
--data '{
    "sort": {
        "direction": "ASC",
        "property": "number_of_tests"
    },
    "pagination": {
        "page": 1,
        "limit": 10
    },
    "filters": [
        {
            "value": "2023-05-21T00:00:00+00:00",
            "filterName": "UNTIL"
        }
    ]
}'
```

## Changes
- Updates the queries to remove some of the CTEs in favor of joins
- Run the count queries in parallel

## Checklist

### Testing

### Process

## Linked Issues
- #9411 

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
